src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/osScanner.hpp

```diff
@@ -13,27 +13,294 @@
 #define _OS_SCANNER_HPP
 
 #include "chainOfResponsability.hpp"
+#include "databaseFeedManager.hpp"
 #include "scanContext.hpp"
+#include "scannerHelper.hpp"
+#include "versionMatcher/versionMatcher.hpp"
 
 /**
  * @brief OsScanner class.
  *
  */
-class OsScanner final : public AbstractHandler<std::shared_ptr<ScanContext>>
+template<typename TDatabaseFeedManager = DatabaseFeedManager,
+         typename TScanContext = ScanContext,
+         typename TSocketDBWrapper = SocketDBWrapper>
+class TOsScanner final : public AbstractHandler<std::shared_ptr<TScanContext>>
 {
+private:
+    std::shared_ptr<TDatabaseFeedManager> m_databaseFeedManager;
+
 public:
+    /**
+     * @brief OsScanner constructor.
+     *
+     * @param databaseFeedManager Database feed manager.
+     */
+    explicit TOsScanner(std::shared_ptr<TDatabaseFeedManager> databaseFeedManager)
+        : m_databaseFeedManager(std::move(databaseFeedManager))
+    {
+    }
     /**
      * @brief Handles request and passes control to the next step of the chain.
      *
      * @param data Scan context.
      * @return std::shared_ptr<ScanContext> Abstract handler.
      */
     // LCOV_EXCL_START
-    std::shared_ptr<ScanContext> handleRequest(std::shared_ptr<ScanContext> data) override
+    std::shared_ptr<ScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
     {
+        nlohmann::json responseHotfixes;
+        static TSocketDBWrapper wdbSocketWrapper(WDB_SOCKET);
+
+        if (data->osPlatform() == "windows")
+        {
+            try
+            {
+                wdbSocketWrapper.query(
+                    WazuhDBQueryBuilder::builder().agentGetHotfixesCommand(data->agentId().data()).build(),
+                    responseHotfixes);
+            }
+            catch (const std::exception& e)
+            {
+                throw OsDataException(e.what());
+            }
+        }
+
+        const auto osCPE = ScannerHelper::parseCPE(data->osCPEName().data());
+
+        auto vulnerabilityScan =
+            [&](const std::string& cnaName, const NSVulnerabilityScanner::ScanVulnerabilityCandidate& callbackData)
+        {
+            try
+            {
+                VersionObjectType objectType = VersionObjectType::DPKG;
```

Comment:
By default, we now assume that the probability of `VersionObjectType::DPKG` is greater, right?

Reply:
yes it is correct.